applications/StructuralMechanicsApplication/python_scripts/automatic_rayleigh_parameters_computation_process.py

```diff
@@ -0,0 +1,181 @@
+            if kratos_utils.IsApplicationAvailable("ExternalSolversApplication"):
+                from KratosMultiphysics import ExternalSolversApplication
+            if kratos_utils.IsApplicationAvailable("EigenSolversApplication"):
+                from KratosMultiphysics import EigenSolversApplication
+        else:
+            raise Exception("ExternalSolversApplication or EigenSolversApplication not available")
+
+        # The general damping ratios
+        damping_ratio_0 = self.settings["damping_ratio_0"].GetDouble()
+        damping_ratio_1 = self.settings["damping_ratio_1"].GetDouble()
+
+        # We get the model parts which divide the problem
+        current_process_info = self.main_model_part.ProcessInfo
+        existing_computation = current_process_info.Has(SMA.EIGENVALUE_VECTOR)
+
+        # Create auxiliar parameters
+        compute_damping_coefficients_settings = KM.Parameters("""
+        {
+            "echo_level"          : 0,
+            "damping_ratio_0"     : 0.0,
+            "damping_ratio_1"     : -1.0,
+            "eigen_values_vector" : [0.0]
+        }
+        """)
+
+        # Setting custom parameters
+        compute_damping_coefficients_settings["echo_level"].SetInt(self.settings["echo_level"].GetInt())
+        compute_damping_coefficients_settings["damping_ratio_0"].SetDouble(damping_ratio_0)
+        compute_damping_coefficients_settings["damping_ratio_1"].SetDouble(damping_ratio_1)
+
+        # We check if the values are previously defined
+        properties = self.main_model_part.GetProperties()
+        for prop in properties:
+            if prop.Has(SMA.SYSTEM_DAMPING_RATIO):
+                self.settings["damping_ratio_0"].SetDouble(prop.GetValue(SMA.SYSTEM_DAMPING_RATIO))
+                break
+        for prop in properties:
+            if prop.Has(SMA.SECOND_SYSTEM_DAMPING_RATIO):
+                self.settings["damping_ratio_1"].SetDouble(prop.GetValue(SMA.SECOND_SYSTEM_DAMPING_RATIO))
+                break
+
+        # We have computed already the eigen values
+        current_process_info = self.main_model_part.ProcessInfo
+        precomputed_eigen_values = self.settings["eigen_values_vector"].GetVector()
+        if len(precomputed_eigen_values) > 1:
+            compute_damping_coefficients_settings["eigen_values_vector"].SetVector(precomputed_eigen_values)
+        else:
+            # If not computed eigen values already
+            if not existing_computation:
+                KM.Logger.PrintInfo("::[MechanicalSolver]::", "EIGENVALUE_VECTOR not previously computed. Computing automatically, take care")
+                from KratosMultiphysics import eigen_solver_factory
+                eigen_linear_solver = eigen_solver_factory.ConstructSolver(self.settings["eigen_system_settings"])
+                builder_and_solver = KM.ResidualBasedBlockBuilderAndSolverWithConstraints(eigen_linear_solver)
```

Comment:
why cannot you use this B&S to setup the system again?

Reply:
Because I need to know the computing model part (which I know we want to deprecate, but that is not the current case)

Reply:
Additionally the equations Ids depend of the B&S used

Reply:
Ok I see

last request then, can you please make this depending on a variable? => see suggestion in other comment (wait for it...)